scripts/extract.py

```diff
@@ -44,22 +45,31 @@ def process(self):
         else:
             self.extract_single_process()
 
-        self.alignments.write_alignments(self.faces.faces_detected)
-
+        self.write_alignments()
         images, faces = Utils.finalize(self.images.images_found,
                                        self.faces.num_faces_detected,
                                        self.faces.verify_output)
         self.images.images_found = images
         self.faces.num_faces_detected = faces
 
+    def write_alignments(self):
+        self.alignments.write_alignments(self.faces.faces_detected)
+
+
     def extract_single_process(self):
         """ Run extraction in a single process """
+        frame_no = 0
         for filename in tqdm(self.images.input_images, file=sys.stdout):
             filename, faces = self.process_single_image(filename)
             self.faces.faces_detected[os.path.basename(filename)] = faces
+            frame_no += 1
+            if frame_no == self.save_interval:
```

Comment:
As per earlier comment, self.save_interval will not necessarily exist, so reference it directly from arguments and make sure it is a passed argument: `if hasattr(self.args, "save_interval") and self.args.save_interval == frame_no`